virtual_rainforest/models/animals/animal_communities.py

```diff
@@ -93,3 +99,31 @@ def birth(self, cohort: AnimalCohort) -> AnimalCohort:
             AnimalCohort(cohort.functional_group, cohort.mass, 0.0)
         )
         return self.cohorts[cohort.name][-1]
+
+    def forage_community(self) -> None:
+        """This function needs to handle the foraging of animal cohorts.
+
+        It should loop over every animal cohort in the community and call either
+        herbivory or predation functions. This will sooner be expanded to include
+        functions for handling scavenging and soil consumption behaviors specifically.
+
+
+        """
+        plant_list = [self.plant_community]
+        carcass_pool = self.carcass_pool
+        soil_pool = self.soil_pool
+
+        animal_list = []  # List to collect all animals
+
+        for functional_group, cohorts in self.cohorts.items():
+            animal_list.extend(cohorts)  # Add cohorts from the current functional group
+
+        for cohort in animal_list:
+            other_animals = [animal for animal in animal_list if animal != cohort]
```

Comment:
You're iterating over a list, so you just use indexing here to omit the target cohort rather than a more expensive list comprehension and comparison:
```suggestion
        for idx, cohort in enumerate(animal_list):
            other_animals = animal_list[0:idx] + animal_list[idx+1:]
```

Reply:
Having said that doesn't this prevent within-cohort cannibalism?

Reply:
> Having said that doesn't this prevent within-cohort cannibalism?

It was intended to prevent within-cohort cannibalism but not within-fg cannibalism. The situation where you are predating on an individual that is both your "species" and identical body-mass would be extremely rare. Maybe I should build something in where this occurs when there are no other remaining animal cohorts.

Reply:
With your above suggestion for a self.prey_groups, all of this will be moot i think. I'll be able to specify cannibalism and its conditions clearly in that attribute.